test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -55,4 +55,42 @@ Describe "Set-Location" -Tags "CI" {
             (Get-Location).Path | Should Be (Get-PSProvider FileSystem).Home
         }
     }
+
+    Context 'Set-Location with last location history' {
+        
+            It 'Should go to last location when specifying minus as a path' {
+                $initialLocation = Get-Location
+                Set-Location ([System.IO.Path]::GetTempPath())
+                Set-Location -
+                (Get-Location).Path | Should Be ($initialLocation).Path
+            }
+
+            It 'Should go to last location when specifying minus as a path using alias' {
+                $initialLocation = Get-Location
+                Set-Location ([System.IO.Path]::GetTempPath())
+                cd -
+                (Get-Location).Path | Should Be ($initialLocation).Path
+            }
+    
+            It 'Should go back to original location when specifying minus as a path twice' {
+                $tempPath = ([System.IO.Path]::GetTempPath())
+                Set-Location $tempLocation
+                $tempLocation = Get-Location
+                Set-Location -
+                Set-Location -
+                (Get-Location).Path | Should Be ($tempLocation).Path
+            }
+
+            It 'Should fail if there is no last location' {
```

Comment:
I'm also surprised about this test - it seems like you need a way to clear the stack before running `cd -` to make sure it reliably raises the exception.

Reply:
I rewrote this test since it does not make sense any more (it was based on the first environment variable based implementation)